### PR TITLE
Create Subject Registration Mono Repo With Frontend and Backend

### PR DESCRIPTION
**Pull request by hanjul**
I have implemented the monorepo according to the requirements in #12. Additionally, the `docker` directory contains `docker-compose.yml` files to bootstrap a development environment.

* Backend dev
```sh
docker-compose up -d
```

* Frontend dev
```sh
docker-compose -f docker-compose.yml -f docker-compose.frontend-dev.yml up -d
```

I also edited the `hsa-realm.json` used in `keycloak` to provide a test user for developers. Though I am unsure about the consequences of the changes. Are there any restrictions on the configuration of the `keycloak` instance in production?

Additionally, I'd appreciate a review of the pipeline. Does the separation into a `test` and `release` pipeline make sense? How should I handle pull requests and tagging in a release? Should dev builds get their own release?

### CODE REVIEW

**Comment by dominikhorn93** [comment]
"Are there any restrictions on the configuration of the keycloak instance in production?"
There are no restrictions. We will have a different realm later, with different admin users and maybe a different theme.
But: all this can be done via mounts and environment variables.


„Additionally, I'd appreciate a review of the pipeline. Does the separation into a test and release pipeline make sense?“
Makes totally sense to me.


„How should I handle pull requests and tagging in a release? Should dev builds get their own release?“
In my point of view you should enable the release pipeline on the main branch too. We are using the following Action to calculate tags: https://github.com/docker/metadata-action

**Comment by dominikhorn93** [comment]
Could you also add the ISSUE_TEMPLATES again? Great for new issues / bugs 😄 
Moreover we should create Templates for PRs. (Definition of Done) -> but that is another issue

File: .github/workflows/release-frontend.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+name: "Release Frontend"
+
+on:
+  push:
+    branches:
+      - 'dev'
```
Comment (dominikhorn93) [inline .github/workflows/release-frontend.yaml:6]: add main

File: .github/workflows/release-frontend.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    tags:
+      - 'v*.*.*'
+    paths:
+      - 'frontend/**'
+  pull_request:
+    branches:
+      - 'master'
```
Comment (dominikhorn93) [inline .github/workflows/release-frontend.yaml]: change to main

File: .github/workflows/release-frontend.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            type=semver,pattern={{major}}
+            type=sha
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Setup NodeJS 14.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.x
+          registry-url: 'https://npm.pkg.github.com'
+          scope: '@flowsquad'
```
Comment (dominikhorn93) [inline .github/workflows/release-frontend.yaml]: remove scope

File: .github/workflows/release-backend.yaml [hunk 4]
```diff
@@ -1,18 +1,24 @@
-name: ci
+name: "Release Backend"
 
 on:
   push:
     branches:
-      - '**'
+      - 'dev'
```
Comment (dominikhorn93) [inline .github/workflows/release-backend.yaml:6]: add main

File: .github/workflows/release-backend.yaml [hunk 5]
```diff
@@ -1,18 +1,24 @@
...
     tags:
       - 'v*.*.*'
+    
+    paths:
+      - 'backend/**'
   pull_request:
     branches:
       - 'master'
```
Comment (dominikhorn93) [inline .github/workflows/release-backend.yaml]: change to main

**Comment by hanjul** [comment]
> Could you also add the ISSUE_TEMPLATES again? Great for new issues / bugs 😄 Moreover we should create Templates for PRs. (Definition of Done) -> but that is another issue

The `ISSUE_TEMPLATE`'s were added after I pulled the repo, thus they were not included. I think they should automatically be part of the merge.

**Review by dominikhorn93 — APPROVED** [review]

**Pull request merged**